File: services/analysis/src/analysis/health.py
"""Health reporting for the analysis service."""

from __future__ import annotations

from dataclasses import asdict, dataclass

from analysis import __version__


@dataclass(frozen=True)
class Status:
    """Service health state reported over HTTP and (later) gRPC."""

    service: str
    status: str
    version: str


def report() -> dict[str, str]:
    """Return the current health status as a serializable mapping."""
    return asdict(
        Status(service="analysis", status="ok", version=__version__)
    )

File: services/replay-engine/internal/health/health.go
// Package health provides liveness/readiness reporting for the replay engine.
package health

// Status is the service health state reported over HTTP and (later) gRPC.
type Status struct {
	Service string `json:"service"`
	Status  string `json:"status"`
	Version string `json:"version"`
}

// Report returns the current health status for the given version.
func Report(version string) Status {
	return Status{
		Service: "replay-engine",
		Status:  "ok",
		Version: version,
	}
}
